### PR TITLE
Add action types

### PR DESCRIPTION
**Pull request by LeoColman**
Github Actions YAML usually isn't as useful as it could be.

https://github.com/krzema12/github-actions-typing

This project setup a way to define types for github actions inputs to build a type-safe DSL.

This commit implements both the typing for this action and the workflow to verify it.

### CODE REVIEW

**Comment by LeoColman** [comment]
Ping @Entrostat-Github can I get some feedback on this one :)

**Comment by LeoColman** [comment]
It shouldn't.

This action-types enables tools that operate on Github Actions to have more information on the action

https://github.com/krzema12/github-actions-kotlin-dsl

This project, for example, can take the types in `action-types` and generate type-safe Kotlin DSL.

----

This should be an addition only, without impact to current users.

**Comment by LeoColman** [comment]
@Kerren-Entrostat 
Do you need help to understand this?
It would be very cool to have this action supported :)

**Comment by krzema12** [comment]
@Kerren-Entrostat do you need any further help? I'm happy to address any questions and accept feedback.

**Comment by Kerren-Entrostat** [comment]
Hi guys, I've been on the fence with this one, the problem is that I want to keep the dependencies at a minimum to ensure that security is maximised.

I'd like to minimise the attack vector to ensure users feel comfortable using this package.

If I could have a bit of time to review the source code of your action and potentially lock in a specific version and only update on review, I will take it under consideration. But I'd like to consult with other users to see how they feel about integrating with external dependencies.

Please know that this is nothing against your hard work and I really think your action is awesome and one that should be used in as many use cases as possible!

**Comment by krzema12** [comment]
@Kerren-Entrostat sure, I understand! Just to be clear: the validation action is executed only in your repository, it doesn't get propagated to consumers of your action.

**Comment by Kerren-Entrostat** [comment]
@krzema12 That changes things a lot :smile: I'm very happy to have a look at this then, thank you for the contribution!

Would it be possible to update the branch with the latest `develop` and then I'll merge it in?

Apologies for the misunderstanding here!

File: .github/workflows/action-types.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  validate-typings:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: krzema12/github-actions-typing@v0
```
Comment (krzema12) [inline .github/workflows/action-types.yml:13]: After recent changes, it should say

```suggestion
      - uses: typesafegithub/github-actions-typing@v1
```

**Pull request merged**